gitlab/mixins.py

```diff
@@ -618,9 +618,8 @@ class AccessRequestMixin(_RestObjectBase):
     _updated_attrs: Dict[str, Any]
     manager: base.RESTManager
 
-    @cli.register_custom_action(
-        ("ProjectAccessRequest", "GroupAccessRequest"), (), ("access_level",)
-    )
+    @cli.register_custom_action(("ProjectAccessRequest", "GroupAccessRequest"))
+    @base.custom_attrs(optional=("access_level",))
```

Comment:
Have we considered having the wrapper figuring these out automatically? Is it possible using introspection to determine the function signature?

Reply:
This might be a lean way to handle it, but we'd lose two things:
* no way to know what args should go into the payload vs. passed on to `http_*` method kwargs. This was the main reason I want to them out from function code into a decorator e.g. for required args.
* we'd lose the args in `--help` for that specific method on the CLI :blush: Not a big deal since I realize I'm claiming here this is being decoupled, but we'd need to be able to pass arbitrary args to the CLI (e.g. https://stackoverflow.com/questions/37367331/is-it-possible-to-use-argparse-to-capture-an-arbitrary-set-of-optional-arguments, though I tried quickly and it'd need more work for us).

Reply:
Hmm, I played around with allowing arbitrary CLI args, and it would have to be added to the actual class subparsers which seems a bit convoluted. So I think I need to drop that idea until we use a better CLI library :)

Reply:
> Hmm, I played around with allowing arbitrary CLI args, and it would have to be added to the actual class subparsers which seems a bit convoluted. So I think I need to drop that idea until we use a better CLI library :)

Sorry, I'm a bit confused. Drop which idea? This whole PR or my idea about a wrapper that figures out the arguments or some other idea?

Reply:
The idea to automatically figure out arguments, yes. We still need these currently for the CLI to work, because `argparse` does not allow arbitrary arguments (without hacking too much) - so we have to store them somewhere.

Also I thought it would be useful in the API portion for when we need to distinguish which arguments in the function we should send as data or not, e.g. to replace this kind of custom logic:
https://github.com/python-gitlab/python-gitlab/blob/387a14028b809538530f56f136436c783667d0f1/gitlab/v4/objects/merge_requests.py#L381-L386